client/landing/stepper/declarative-flow/hosting-site-creation-flow.ts

```diff
@@ -60,6 +84,20 @@ const hosting: Flow = {
 
 		return { submit };
 	},
+	useSideEffect( currentStepSlug ) {
+		const { resetOnboardStore } = useDispatch( ONBOARD_STORE );
+
+		useEffect(
+			() => {
+				if ( currentStepSlug === 'options' ) {
+					resetOnboardStore();
+				}
```

Comment:
This does something when the user navigates directly to `/setup/new-hosted-site/options`, but not when the user goes to `/setup/new-hosted-site` (which is the usual URL we use to start a flow). That's because `currentStepSlug === undefined`.

In my experience, deciding when is the best time to reset the flow is one of the trickiest parts of the onboarding process. No easy answers 🤷 Reseting the flow for each fresh load of the `options` step seems a good place to start though.

The `free` flow does it [for every page load](https://github.com/Automattic/wp-calypso/blob/51fb46a0166b32b40ef986bc5d898b13775c3233/client/landing/stepper/declarative-flow/free.ts#L38) but that seems too aggressive for me. If the user refreshes on the plans step we should retain their geo affinity etc.